### PR TITLE
BFD-1731: Import bfd-server dashboards

### PR DESCRIPTION
**Pull request by garfield786**
**JIRA Ticket:**
[BFD-1731](https://jira.cms.gov/browse/BFD-1731)

**User Story or Bug Summary:**
Importing Existing BFD AWS Cloud Watch Dashboards in Terraform

---

### What Does This PR Do?

Adding New FHIR AWS CW Dashboards

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/modules/stateful/main.tf [hunk 1]
```diff
@@ -122,6 +122,7 @@ module "aurora" {
 ## S3 Buckets
 #
 
+
```
Comment (mjburling) [inline ops/terraform/modules/stateful/main.tf:125]:
```suggestion
```

File: ops/terraform/env/test/stateless/main.tf [hunk 2]
```diff
@@ -20,4 +20,10 @@ module "stateless" {
   ssh_key_name    = var.ssh_key_name
   git_branch_name = var.git_branch_name
   git_commit_id   = var.git_commit_id
+
```
Comment (mjburling) [inline ops/terraform/env/test/stateless/main.tf:23]:
```suggestion
```

File: ops/terraform/env/test/stateful/main.tf [hunk 3]
```diff
@@ -44,4 +44,5 @@ module "stateful" {
   victor_ops_url = var.victor_ops_url
 
   medicare_opt_out_config = var.medicare_opt_out_config
+
```
Comment (mjburling) [inline ops/terraform/env/test/stateful/main.tf:47]:
```suggestion
```

File: ops/terraform/env/prod/stateful/main.tf [hunk 4]
```diff
@@ -47,4 +47,5 @@ module "stateful" {
   victor_ops_url = var.victor_ops_url
 
   medicare_opt_out_config = var.medicare_opt_out_config
+
```
Comment (mjburling) [inline ops/terraform/env/prod/stateful/main.tf:50]:
```suggestion
```

File: ops/terraform/env/prod-sbx/stateful/main.tf [hunk 5]
```diff
@@ -44,4 +44,5 @@ module "stateful" {
   victor_ops_url = var.victor_ops_url
 
   medicare_opt_out_config = var.medicare_opt_out_config
+
```
Comment (mjburling) [inline ops/terraform/env/prod-sbx/stateful/main.tf:47]:
```suggestion
```

File: ops/terraform/modules/stateless/variables.tf [hunk 6]
```diff
@@ -33,3 +33,13 @@ variable "is_public" {
...
 }
+
+variable "dashboard_name" {
+  description = "Name of the bfd cloudwatch dashboards"
+  type = string
+}
+
+variable "dashboard_namespace" {
+  description = "The namespace in which the dashboards live in"
+  type = string
```
Comment (mjburling) [inline ops/terraform/modules/stateless/variables.tf:44]: Could you be sure to run `terraform fmt` in this module and elsewhere? The alignment's out of order here.

File: ops/terraform/modules/resources/bfd_cw_dashboards/variables.tf [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  type = string
+}
+
+variable "dashboard_namespace" {
+  type = string
+}
+
+variable "asg" {
+    type = string
+}
```
Comment (mjburling) [inline ops/terraform/modules/resources/bfd_cw_dashboards/variables.tf:11]: Can you add descriptions to your input variables here?

**Review by mjburling — CHANGES_REQUESTED** [review]
There are a few minor whitespace issues here and there, some missing descriptions for variables, and a need for `terraform fmt`. You ought to drop the `ops/terraform/modules/resources/bfd_cw_dashboards/templates/bfd-dashboards.json`–it's unsued since you introduced `ops/terraform/modules/resources/bfd_cw_dashboards/templates/bfd-dashboards.tpl`, right?

Otherwise, I think we can go ahead with this. Thanks!

**Review by mjburling — APPROVED** [review]
OK, @garfield786. This is a step in the right direction and a fine place from which we can iterate. Let's see a deployment through with this inside. Thanks!

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**